plugins/q/src/data/extractor-s.js

```diff
@@ -97,74 +97,76 @@ function cellToValue({
 export default function extract(config, dataset, cache, util) {
   const cfgs = Array.isArray(config) ? config : [config];
   let dataItems = [];
-  cfgs.forEach((cfg) => {
-    if (typeof cfg.field !== 'undefined') {
+  for (let i = 0; i < cfgs.length; i++) {
+    if (typeof cfgs[i].field !== 'undefined') {
       const cube = dataset.raw();
       const sourceKey = dataset.key();
-      const f = typeof cfg.field === 'object' ? cfg.field : dataset.field(cfg.field);
-      const { props, main } = util.normalizeConfig(cfg, dataset);
+      const f = typeof cfgs[i].field === 'object' ? cfgs[i].field : dataset.field(cfgs[i].field);
+      const { props, main } = util.normalizeConfig(cfgs[i], dataset);
       const propsArr = Object.keys(props);
 
-      const track = !!cfg.trackBy;
-      const trackType = typeof cfg.trackBy;
+      const track = !!cfgs[i].trackBy;
+      const trackType = typeof cfgs[i].trackBy;
       const tracker = {};
       const trackedItems = [];
       const items = [];
 
-      cube.qDataPages.forEach((page) => {
-        const fn = getFieldAccessor(f, page, { cache });
+      for (let j = 0; j < cube.qDataPages.length; j++) {
+        const fn = getFieldAccessor(f, cube.qDataPages[j], { cache });
         if (fn === -1) {
-          return;
+          continue;
```

Comment:
The return did not do anything in the for each loop, so I replaced it with continue instead. If there is a better case here I that missed, please let me know.